ADR/0011-roles-and-permissions.md

```diff
@@ -58,6 +59,7 @@ We will use the built-in Kubernetes RBAC system for AppStudio's role and permiss
 |               | *Service Access*        | appstudio.redhat.com      | get, list, watch, create, update, patch         | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spifilecontentrequests, spiaccesstokendataupdates
 |               | *Remote Secrets*        | appstudio.redhat.com      | get, list, watch                                | remotesecrets
 |               | Build Service           | appstudio.redhat.com      | get, list, watch, create                        | buildpipelineselectors
+|               | Project Controller      | projctl.konflux.dev       | get, list, watch, create, update, patch, delete | projects, projectdevelopmentstreams, projectdevelopmentstreamtemplates
```

Comment:
Do you want to grant `delete` permissions to maintainers? Maintainers do not have those permissions for Components or Applications.

Reply:
We did grant them in practice for now, so might as well have the docs reflect that. We can consider changing it later.

Reply:
If that is the current state, would you mind including that update into this PR as well since it relates to the new permissions added?

Reply:
@arewm what changes should I make, if any?

Reply:
@arewm no, this PR reflects what the permissions for the project controller are ATM. Thos are, as you noted, slightly different the the permissions for Applications and Components